packages/core/src/components/fields/ObjectField.tsx

```diff
@@ -68,127 +117,143 @@ class ObjectField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends Fo
    * @param addedByAdditionalProperties - Flag indicating whether this property is an additional property
    * @returns - The onPropertyChange callback for the `name` property
    */
-  onPropertyChange = (name: string, addedByAdditionalProperties = false) => {
-    return (value: T | undefined, path: FieldPathList, newErrorSchema?: ErrorSchema<T>, id?: string) => {
-      const { onChange } = this.props;
+  const onPropertyChange = useCallback(
+    (value: T | undefined, path: FieldPathList, newErrorSchema?: ErrorSchema<T>, id?: string) => {
       if (value === undefined && addedByAdditionalProperties) {
-        // Don't set value = undefined for fields added by
-        // additionalProperties. Doing so removes them from the
-        // formData, which causes them to completely disappear
-        // (including the input field for the property name). Unlike
-        // fields which are "mandated" by the schema, these fields can
-        // be set to undefined by clicking a "delete field" button, so
-        // set empty values to the empty string.
+        // Don't set value = undefined for fields added by additionalProperties. Doing so removes them from the
+        // formData, which causes them to completely disappear (including the input field for the property name). Unlike
+        // fields which are "mandated" by the schema, these fields can be set to undefined by clicking a "delete field"
+        // button, so set empty values to the empty string.
         value = '' as unknown as T;
       }
       onChange(value, path, newErrorSchema, id);
-    };
-  };
+    },
+    [onChange, addedByAdditionalProperties],
+  );
 
-  /** Returns a callback to handle the onDropPropertyClick event for the given `key` which removes the old `key` data
-   * and calls the `onChange` callback with it
-   *
-   * @param key - The key for which the drop callback is desired
-   * @returns - The drop property click callback
+  /** The key change event handler; Called when the key associated with a field is changed for an additionalProperty.
+   * simply returns a function that call the `handleKeyChange()` event with the value
    */
-  onDropPropertyClick = (key: string) => {
-    return (event: DragEvent) => {
-      event.preventDefault();
-      const { onChange, formData, fieldPathId } = this.props;
-      const copiedFormData = { ...formData } as T;
-      unset(copiedFormData, key);
-      // drop property will pass the name in `path` array
-      onChange(copiedFormData, fieldPathId.path);
-    };
-  };
+  const onKeyRename = useCallback(
+    (value: string) => {
+      setWasPropertyKeyModified(propertyName !== value);
```

Comment:
Seems like this yields a small change in behavior

Before: once `wasPropertyKeyModified` was true, it would remain true for the lifetime of the component. I don't think it could ever be set back to `false`.

After: if this callback is invoked without changing the `propertyName`, it could go from `true` to `false`.

Not sure what the actual impact of this change is

Reply:
I can roll it back to what was there before